BUILDING.md

```diff
@@ -1,47 +1,65 @@
 # Building with CMake
 
+This project encourages uses of up-to-date build tools and contemporary development environments.
+To build from sources, you need:
+
+- CMake 3.26 or higher
+- A C++ compiler that supports C++23 (or equivalent) which is just a bug fix of C++20
+- An implementation of the C++ Core Guidelines Support Library (GSL)
+
 ## Dependencies
 
 For a list of dependencies, please refer to [vcpkg.json](vcpkg.json).
 
+This project depends on the [Microsoft implementation of GSL](https://github.com/microsoft/GSL), which is available on all supported platforms.  We recommend that you use your favorite package manager to install it prior to building and installing the IFC SDK from source.  If you just want to build the project and try it without installing it, then there is no need to install the GSL, but in that case the build step is a bit more involved.
+
+On Windows platforms, we recommend using [vckpg](https://vcpkg.io/en/getting-started.html) as the C++ package manager.  Issue
+```sh
+$ ./vckpg install --triplet x64-windows ms-gsl
+```
+in the VCPKG directory, to install the GSL.  The above command assumes you are building for the `x64` target.  Replace that with the appropriate triplet if you are taregting a different platform like `arm64`.
+
+On linux platforms, there is no commonly agreed-upon name for that package.  Here are a few common name for the Microsoft implementation of the GSL on some linux distributions if you're using the system package manager:
+
+- arch: `microsoft-gsl`
+- debian: `ms-gsl`
+- fedora: `guidelines-support-library`
+- gentoo: `dev-cpp/ms-gsl`
+- ubuntu: `ms-gsl`
+
+Please, consult your linux distribution manual to find the appropriate package name.
```

Comment:
This might be better to upstream to the GSL readme and just refer to that instead.

Reply:
> This might be better to upstream to the GSL readme and just refer to that instead.

This project as a dependency on a component and it is providing information in how to satisfy that dependency.  I would like to have the instructions for building this project to be approachable and relatively self contained to get started.  If some of the information here could be also useful somewhere else, that is great, but that shouldn't prevent us from making it easy to quickly get started.